cms/djangoapps/bundles/views.py

```diff
@@ -45,6 +57,31 @@ class to hold user state.
     return get_user_state_kvs._kvs
 
 
+def get_xblock_handler_url(usage_id_str, handler_name, suffix, user_id):
```

Comment:
If we need to add this type of auth to other URLs, we will have to create a new view for each? Can this not be moved into a decorator etc? Curious if there are recommended patterns this setup is following?

Reply:
I'm not aware of any other URLs where we'll need this. It's vaguely similar to [S3 Query String Authentication](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html).